exporter/collector/metrics.go

```diff
@@ -746,12 +774,180 @@ func (m *metricMapper) normalizeNumberDataPoint(point pdata.NumberDataPoint, ide
+				zap.String("lastRecordedReset", start.Timestamp().String()),
+				zap.String("dataPoint", point.Timestamp().String()),
+			)
+			return nil
+		}
+		// Make a copy so we don't mutate underlying data
+		newPoint := pdata.NewHistogramDataPoint()
+		point.CopyTo(newPoint)
+		// Use the start timestamp from the normalization point
+		newPoint.SetStartTimestamp(start.Timestamp())
+		// Adjust the value based on the start point's value
+		newPoint.SetCount(point.Count() - start.Count())
+		// We drop points without a sum, so no need to check here.
+		newPoint.SetSum(point.Sum() - start.Sum())
+		pointBuckets := point.BucketCounts()
+		startBuckets := start.BucketCounts()
+		if !bucketBoundariesEqual(point.ExplicitBounds(), start.ExplicitBounds()) {
+			// The number of buckets changed, so we can't normalize points anymore.
+			// Treat this as a reset by recording and dropping this point.
+			m.normalizationCache.SetHistogramDataPoint(identifier, &point)
+			return nil
+		}
+		newBuckets := make([]uint64, len(pointBuckets))
+		for i := range pointBuckets {
+			newBuckets[i] = pointBuckets[i] - startBuckets[i]
+		}
+		newPoint.SetBucketCounts(newBuckets)
+		normalizedPoint = &newPoint
+	}
+	if (!ok && point.StartTimestamp() == 0) || !point.StartTimestamp().AsTime().Before(point.Timestamp().AsTime()) {
+		// This is the first time we've seen this metric, or we received
+		// an explicit reset point as described in
+		// https://github.com/open-telemetry/opentelemetry-specification/blob/9555f9594c7ffe5dc333b53da5e0f880026cead1/specification/metrics/datamodel.md#resets-and-gaps
+		// Record it in history and drop the point.
+		m.normalizationCache.SetHistogramDataPoint(identifier, &point)
+		return nil
+	}
+	return normalizedPoint
+}
+
+func bucketBoundariesEqual(a, b []float64) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// normalizeExponentialHistogramDataPoint returns a point which has been normalized against an initial
+// start point, or nil if the point should be dropped.
+func (m *metricMapper) normalizeExponentialHistogramDataPoint(point pdata.ExponentialHistogramDataPoint, identifier string) *pdata.ExponentialHistogramDataPoint {
+	// if the point doesn't need to be normalized, use original point
+	normalizedPoint := &point
+	start, ok := m.normalizationCache.GetExponentialHistogramDataPoint(identifier)
+	if ok {
+		if !start.StartTimestamp().AsTime().Before(point.Timestamp().AsTime()) {
+			// We found a cached start timestamp that wouldn't produce a valid point.
+			// Drop it and log.
+			m.obs.log.Info(
+				"data point being processed older than last recorded reset, will not be emitted",
+				zap.String("lastRecordedReset", start.Timestamp().String()),
+				zap.String("dataPoint", point.Timestamp().String()),
+			)
+			return nil
+		}
+		if point.Scale() != start.Scale() {
+			// TODO: It is possible, but difficult to compare exponential
```

Comment:
This is "ok" for now.

I'd like to open abug that does the following:


1. If the "start" histogram scale is "lower resolution", then rescale the newest point to the start histogram scale and normalize.
2. If the "start" histogram scale is "higher resolution" then rescale the start point to the new resolution and REPLACE the cache with this rescaled version, using this version from now forward.
3. Always ensure scale value is the "lowest resolution" of start, current.

If OTEL hasn't provided a "rescale" method for ExponentialHistogram, it should.

Side note:  The "offset" for buckets shouldn't be as important, but MAY matter if we need to lower resolution even further.  The crux of the algorithm should remain the same.

Reply:
Opened https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/366